button/stories/button.stories.tsx

```diff
@@ -11,40 +11,50 @@ export default meta;
 
 export const Playground: StoryObj<ButtonProps> = {
   argTypes: {
-    children: { type: { name: 'string', required: true }, control: 'text' },
+    children: { type: { name: 'string', required: true }, control: 'text', defaultValue: 'Button' },
     variant: {
-      type: { name: 'string', required: true },
-      control: 'select',
+      type: 'string',
+      control: 'radio',
       options: ['primary', 'secondary'],
     },
-    compact: { type: { name: 'boolean', required: true }, control: 'boolean' },
-    disabled: { type: { name: 'boolean', required: true }, control: 'boolean' },
+    compact: { type: 'boolean', control: 'boolean' },
+    disabled: { type: 'boolean', control: 'boolean' },
+    icon: {
+      type: 'string',
+      options: iconNames,
+      control: 'select',
+    },
     iconPosition: {
-      type: { name: 'string', required: true },
-      options: ['none', 'left', 'right'],
-      control: { type: 'select' },
+      if: { arg: 'icon' },
+      type: 'string',
+      control: 'radio',
+      options: ['left', 'right'],
     },
-    icon: {
-      type: { name: 'string', required: false },
-      options: [iconNames[0]],
-      control: {
-        type: 'select',
-        labels: {
-          undefined: 'none',
-          left: 'left',
-          right: 'right',
+    loading: { type: 'boolean', control: 'boolean' },
+  },
+  parameters: {
+    s2d: {
+      variantProperties: [
+        'variant',
+        'compact',
+        'disabled',
+        {
+          fromArg: 'icon',
+          values: [iconNames[0]],
```

Comment:
@gluck @georges-gomes I could't find any way to hide this from the matrix and as a result from the Figma component, this is a subcomponent

![image](https://user-images.githubusercontent.com/137844/204649982-f1236e72-b016-45f2-9e51-e1ce6f43978a.png)

Reply:
I know I could just define this as a defaultValue on `argTypes`, but I don't think it's good to require again using Storybook definitions to achieve this goal, as the whole point of `s2d.variantProperties` was to decouple them, so IMO you need to add a way to set default argument values in `s2d.variantsProperties` too, maybe even add `s2d.defaultArgValues` or something like that

Reply:
Yes I thought it was unnecessary as you could already define args in the story, but I kind-of agree about the decoupling, will expose initArgs there (already available when writing the variant definition inside the plugin)

Reply:
I tried this, the icon is not rendered

<img src="https://user-images.githubusercontent.com/137844/205337444-2053cf19-cc17-489f-8214-5049b5bde285.png" width="200">

then tried this, all icons make over button 40k variants, logically

<img src="https://user-images.githubusercontent.com/137844/205347865-a8be9511-6f8f-474f-b771-c23d32de32e9.png" width="200">

tried this for fun, it just breaks

![image](https://user-images.githubusercontent.com/137844/205350225-86dce9e4-0524-4f89-bcb1-e57d9379bb9b.png)

looks like `initArgs` doesn't solve this issue